ugc/storage.cpp

```diff
@@ -72,6 +72,61 @@ string SerializeUGCIndex(vector<Storage::UGCIndex> const & indexes)
   unique_ptr<char, JSONFreeDeleter> buffer(json_dumps(array.get(), JSON_COMPACT | JSON_ENSURE_ASCII));
   return string(buffer.get());
 }
+
+template <typename UGCUpdate>
+Storage::SettingResult SetGenericUGCUpdate(
+    vector<Storage::UGCIndex> & indexes, size_t & numberOfDeleted, FeatureID const & id,
+    UGCUpdate const & ugc,
+    unique_ptr<FeatureType> featureType,
+    Version const version = Version::Latest)
+{
+  if (!ugc.IsValid())
+    return Storage::SettingResult::InvalidUGC;
+
+  auto const mercator = feature::GetCenter(*featureType);
+  feature::TypesHolder th(*featureType);
+  th.SortBySpec();
+  auto const optMatchingType = ftraits::UGC::GetType(th);
+  CHECK(optMatchingType, ());
+  auto const type = th.GetBestType();
+  for (auto & index : indexes)
+  {
+    if (type == index.m_type && mercator == index.m_mercator && !index.m_deleted)
+    {
+      index.m_deleted = true;
+      ++numberOfDeleted;
+      break;
+    }
+  }
+
+  Storage::UGCIndex index;
+  uint64_t offset;
+  if (!GetUGCFileSize(offset))
+    offset = 0;
+
+  index.m_mercator = mercator;
+  index.m_type = type;
+  index.m_matchingType = *optMatchingType;
+  index.m_mwmName = id.GetMwmName();
+  index.m_dataVersion = id.GetMwmVersion();
+  index.m_featureId = id.m_index;
+  index.m_offset = offset;
+
+  auto const ugcFilePath = GetUGCFilePath();
+  try
+  {
+    FileWriter w(ugcFilePath, FileWriter::Op::OP_APPEND);
+    Serialize(w, ugc, version);
+    indexes.emplace_back(move(index));
+  }
+  catch (FileWriter::Exception const & exception)
+  {
+    LOG(LERROR, ("Exception while writing file:", ugcFilePath, "reason:", exception.Msg()));
```

Comment:
s/Msg/what/

Reply:
Could you please explain to me why I should use ```what``` instead of ```Msg```?

Reply:
what() performs normalization of Msg to ASCII.

Reply:
ok, thanks